soroban-sdk-macros/src/derive_enum_int.rs

```diff
@@ -144,15 +143,19 @@ pub fn derive_type_enum_int(
                 type Error = #path::xdr::Error;
                 #[inline(always)]
                 fn try_into(self) -> Result<#path::xdr::ScVal, #path::xdr::Error> {
-                    Ok((*self as u32).into())
+                    Ok(match self {
+                        #(#try_intos,)*
+                    })
```

Comment:
This optimisation doesn't matter, it's in testutils, it doesn't impact wasm builds.